install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -939,10 +939,6 @@ hubble:
   skipUnknownCGroupIDs: ~
 
   peerService:
-    # -- Enable a K8s Service for the Peer service, so that it can be accessed
-    # by a non-local client. This configuration option is deprecated, the peer
-    # service will be non-optional starting Cilium v1.14.
-    enabled: true
```

Comment:
@qmonnet I think CI is failing because removing this Helm value make it so the peer svc cannot be enabled anymore as [the guard check for the k8s hubble-peer svc](https://github.com/cilium/cilium/blob/bcb608d83220489c1d22f849bda70b4678f3e149/install/kubernetes/cilium/templates/hubble/peer-service.yaml#L1) can never evaluate to true. Consequently Hubble Relay can't connect to Hubble server and all Relay CI tests are (rightfully) failing. This helm invocation confirms it

```console
% helm template cilium ./install/kubernetes/cilium --namespace kube-system --set hubble.enabled=true --set hubble.relay.enabled=true | grep hubble-peer
    peer-service: "hubble-peer.kube-system.svc.cluster.local:443"
```

:point_up: here Relay is configured to connect to the hubble-peer svc but the service itself is not part of the manifest generated by Helm. Removing `.Values.hubble.peerService.enabled` from the mentionned guard condition should fix the problem

```patch
diff --git a/install/kubernetes/cilium/templates/hubble/peer-service.yaml b/install/kubernetes/cilium/templates/hubble/peer-service.yaml
index 574c8da7ec..f54fa6c6d2 100644
--- a/install/kubernetes/cilium/templates/hubble/peer-service.yaml
+++ b/install/kubernetes/cilium/templates/hubble/peer-service.yaml
@@ -1,4 +1,4 @@
-{{- if and .Values.agent .Values.hubble.enabled .Values.hubble.peerService.enabled }}
+{{- if and .Values.agent .Values.hubble.enabled }}
 apiVersion: v1
 kind: Service
 metadata:
```

Reply:
Right, I did miss that one. Thanks a lot!